src/Components/Pages/SetWorkLoadPage.js

```diff
@@ -26,10 +31,13 @@ const SetWorkloadPage = ({
   const { isLoading, error, data } = useEusVersions();
 
   useEffect(() => {
-    if (workload.includes('Extended') && data) {
-      setExtendedReleaseProduct(extendedReleaseProduct || data[0].name);
+    if (workload.includes('Extended') && data?.length > 0) {
+      setExtendedReleaseProduct((prev) => prev || data[0]?.name);
       setExtendedReleaseVersion(
-        extendedReleaseVersion || data[0].configurations[0].version
+        (prev) =>
+          prev ||
+          activationKey?.releaseVersion ||
+          data[0]?.configurations[0]?.version
       );
```

Comment:
Question: What does the `(prev) => prev` identity function do?

Reply:
I used prev-prev || value to say if there was already a truthy value it would stay unchanged.if prev is falsy it'll assign the new value . Basically to prevent the state from being overwritten, and allowing it to stay its first assigned value.

Reply:
Oh interesting - I didn't even know you could get the previous version in `setState` - very cool!